Finish Up File Upload Form structure

PraveenMut opened the pull request.

PraveenMut committed:
add this bindings to file upload form

PraveenMut committed:
change onUpload to onFileAdd for better clarity

PraveenMut committed:
create handle upload form change

PraveenMut committed:
create onFileMethod

PraveenMut committed:
fix up on FileAdd this binding in constructor line 22

PraveenMut committed:
add physical structure to File Add Form

PraveenMut committed:
add styling and margins to button

PraveenMut committed:
add place to list uploaded files

PraveenMut committed:
fix up handleUploadChange method

PraveenMut committed:
improve file handling method

PraveenMut committed:
add styling to file upload form

PraveenMut commented:
Here is the File Upload Flow: 
https://www.dropbox.com/s/r99uaplr9qg6m99/upload_flow.gif?dl=0
![upload_flow](https://user-images.githubusercontent.com/45863687/61167961-9d51d800-a589-11e9-85fb-93b219b7b7a0.gif)

en3on commented on src/app/App.js:
```diff
@@ -1,14 +1,12 @@
     return (
     <div>
-
+      <FileUpload />
```
Remove component

en3on commented on src/fileUpload/FileUploadForm.js:
```diff
@@ -0,0 +1,82 @@
+    this.state = {
+      files: [],
+      filenames: [],
+      description: '',
```
Redundant line due to line 47
```suggestion   
```

en3on commented on src/fileUpload/FileUploadForm.js:
```diff
@@ -0,0 +1,82 @@
+      files: [...this.state.files, e.target.files[0]],
+      filenames: [...this.state.filenames, e.target.files[0].name],
+    });
+    e.target.value = '';
```
Redundant due to line 35
```suggestion
```

en3on commented on src/app/App.js:
```diff
@@ -1,14 +1,12 @@
-import ContactForm from '../contact/ContactForm';
-import SignInForm from '../user/SignInForm';
-import RegisterForm from '../user/RegisterForm';
+import FileUpload from '../fileUpload/FileUploadForm';
```
Remove import

en3on reviewed (approved):
Nice stuff man!!

Just a couple of minor things to change, but it looks awesome nevertheless!!

en3on reviewed (changes requested):
Approved by mistake lol

See above review

PraveenMut committed:
remove FileUpload component from App.js

PraveenMut committed:
remove unneeded method

PraveenMut committed:
fix up controlId conflict in line 49, remove id as it is inherited by controlId

PraveenMut commented:
Made some changes, cleaned up the control id. Most importantly, I removed the unncessary fileadd method as the onchange semed way more intuitive. Works pretty well... looks really nice haha.

PraveenMut committed:
remove FileUpload component from app.js

PraveenMut commented:
OOPS LOL, i forgot to remove the FileUpload component from app comp lol. My bad. Fixed.

alex-double-u-old reviewed (approved):
Beautiful

it's a strong yes from me

en3on reviewed (approved):
Perfect!

PraveenMut committed:
Merge branch 'master' into file-upload-form
